### PR TITLE
DeltaLog checks for required Spark configs and throws an error if not found

### PR DESCRIPTION
**Pull request by ganeshchand**
## Description
Resolves https://github.com/delta-io/delta/issues/1144

## How was this patch tested?

- Added a new test in `DeltaLogSuite`: 
``` scala
test("DeltaLog should not throw exception if SparkSession in initialized with " +
    ".withExtension api and spark.sql.catalog.spark_catalog conf is set")

test("DeltaLog should not throw exception if SparkSession in initialized with " +
    ".withExtension api and spark.sql.catalog.spark_catalog conf is set")
```
- Refactored other unit tests

## Does this PR introduce _any_ user-facing changes?

Yes. If  `spark.sql.catalog.spark_catalog` configuration is not found in the `SparkSession`, it now throws the following user friendly error.

```
This Delta operation requires the SparkSession to be configured with the
DeltaSparkSessionExtension and the DeltaCatalog. Please set the necessary
configurations when creating the SparkSession as shown below.

  SparkSession.builder()
    .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
    .config("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog")
    ...
    .getOrCreate()

If you are using spark-shell/pyspark/spark-submit, you can add the required configurations to the command as show below: --conf spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension --conf spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog
```

Signed-off-by: Ganesh Chand<<ganeshchand@gmail.com>>

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 1]
```diff
@@ -81,6 +84,24 @@ class DeltaLog private(
 
   protected def spark = SparkSession.active
 
+  /**
+   * Issue 1144 - Verify the required Spark conf for delta
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:88]: nit: remove `Issue 1144 -`

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 2]
```diff
@@ -81,6 +84,24 @@ class DeltaLog private(
...
+   * `spark.sql.extensions` and `spark.sql.catalog.spark_catalog` config are missing.
+   * If `DeltaSparkSessionExtension` is alternatively activated using the `.withExtension()` API,
+   * we expect the DeltaSqlParser to be available and try parse the vacuum command.
+   * If not successful, we throw exception.
+   */
+  if(spark.conf.getOption(StaticSQLConf.SPARK_SESSION_EXTENSIONS.key)
+    .orElse(Try(spark.sessionState.sqlParser.parsePlan("vacuum delta_table")).toOption)
+    .isEmpty ||
+    spark.conf.getOption(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key).isEmpty) {
+    throw new DeltaRuntimeException(
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:99]: Could you define this in `DeltaErrors`? For this case, it's better to use `DeltaIllegalStateException`.

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 3]
```diff
@@ -81,6 +84,24 @@ class DeltaLog private(
...
+   * Throw `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if both
+   * `spark.sql.extensions` and `spark.sql.catalog.spark_catalog` config are missing.
+   * If `DeltaSparkSessionExtension` is alternatively activated using the `.withExtension()` API,
+   * we expect the DeltaSqlParser to be available and try parse the vacuum command.
+   * If not successful, we throw exception.
+   */
+  if(spark.conf.getOption(StaticSQLConf.SPARK_SESSION_EXTENSIONS.key)
+    .orElse(Try(spark.sessionState.sqlParser.parsePlan("vacuum delta_table")).toOption)
+    .isEmpty ||
+    spark.conf.getOption(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key).isEmpty) {
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:98]: These two configs must be set via static configs. Could you check them in `spark.sparkContext.getConf` instead?

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 4]
```diff
@@ -81,6 +84,24 @@ class DeltaLog private(
 
   protected def spark = SparkSession.active
 
+  /**
+   * Issue 1144 - Verify the required Spark conf for delta
+   * Throw `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if both
+   * `spark.sql.extensions` and `spark.sql.catalog.spark_catalog` config are missing.
+   * If `DeltaSparkSessionExtension` is alternatively activated using the `.withExtension()` API,
+   * we expect the DeltaSqlParser to be available and try parse the vacuum command.
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:92]: > try parse the vacuum command.

Why do we need to try to parse the command?

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 5]
```diff
@@ -258,7 +279,8 @@ class DeltaLog private(
       startVersion: Long,
       failOnDataLoss: Boolean = false): Iterator[(Long, Seq[Action])] = {
     val hadoopConf = newDeltaHadoopConf()
-    val deltas = store.listFrom(deltaFile(logPath, startVersion), hadoopConf).filter(isDeltaFile)
+    val deltas = store.listFrom(deltaFile(logPath, startVersion), hadoopConf)
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:282]: revert unrelated changes

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 6]
```diff
@@ -280,11 +302,11 @@ class DeltaLog private(
       startVersion: Long,
       failOnDataLoss: Boolean = false): Iterator[(Long, FileStatus)] = {
     val deltas = store.listFrom(deltaFile(logPath, startVersion), newDeltaHadoopConf())
-      .filter(isDeltaFile)
+      .filter(f => isDeltaFile(f.getPath))
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:305]: ditto

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 7]
```diff
@@ -280,11 +302,11 @@ class DeltaLog private(
...
     // Subtract 1 to ensure that we have the same check for the inclusive startVersion
     var lastSeenVersion = startVersion - 1
     deltas.map { status =>
-      val version = deltaVersion(status)
+      val version = deltaVersion(status.getPath)
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:309]: ditto

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 8]
```diff
@@ -561,13 +583,11 @@ object DeltaLog extends DeltaLogging {
   private def apply(spark: SparkSession, rawPath: Path, clock: Clock = new SystemClock): DeltaLog =
     apply(spark, rawPath, Map.empty, clock)
 
-
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:564]: ditto

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 9]
```diff
@@ -561,13 +583,11 @@ object DeltaLog extends DeltaLogging {
...
   private def apply(
       spark: SparkSession,
       rawPath: Path,
       options: Map[String, String],
-      clock: Clock
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:569]: ditto

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaErrorsSuite.scala [hunk 10]
```diff
@@ -47,7 +47,7 @@ import org.apache.spark.sql.catalyst.expressions.{AttributeReference, ExprId, Le
 import org.apache.spark.sql.catalyst.expressions.Uuid
 import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
 import org.apache.spark.sql.connector.catalog.Identifier
-import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.internal.{SQLConf, StaticSQLConf}
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/sql/delta/DeltaErrorsSuite.scala:50]: Looks like the changes in this file are not needed?

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaCommitLockSuite.scala [hunk 11]
```diff
@@ -38,6 +41,9 @@ class DeltaCommitLockSuite extends SparkFunSuite with LocalSparkSession with SQL
     spark = SparkSession.builder()
       .config("spark.delta.logStore.class", classOf[AzureLogStore].getName)
       .master("local[2]")
+      .config(StaticSQLConf.SPARK_SESSION_EXTENSIONS.key,
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/sql/delta/DeltaCommitLockSuite.scala:44]: Maybe create a until method in `DeltaTestUtils` to avoid repeating the same config code?

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 12]
```diff
@@ -81,6 +84,24 @@ class DeltaLog private(
...
 
+  /**
+   * Issue 1144 - Verify the required Spark conf for delta
+   * Throw `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if both
+   * `spark.sql.extensions` and `spark.sql.catalog.spark_catalog` config are missing.
+   * If `DeltaSparkSessionExtension` is alternatively activated using the `.withExtension()` API,
+   * we expect the DeltaSqlParser to be available and try parse the vacuum command.
+   * If not successful, we throw exception.
+   */
+  if(spark.conf.getOption(StaticSQLConf.SPARK_SESSION_EXTENSIONS.key)
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:95]: As a best practice, could you add an internal config in DeltaSQLConf to disable the check?

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaLogSuite.scala [hunk 13]
```diff
@@ -505,4 +508,54 @@ class DeltaLogSuite extends QueryTest
       )
     }
   }
+
+  test("DeltaLog should throw exception if spark.sql.extensions and " +
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/sql/delta/DeltaLogSuite.scala:512]: Could you move these tests to `DeltaExtensionAndCatalogSuite.scala`? This would make new tests added after these tests in future work.

**Review by zsxwing — COMMENTED** [review]
Thanks for the contribution! Left a few minor comments

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 1 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:88]: will do.

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 2 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:99]: We already have `def configureSparkSessionWithExtensionAndCatalog(originalException: Throwable)` defined in DeltaErrors. Should we do this then?

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 3 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:98]: will do. We also have a case where a SparkSession is initialized with `.withExtensions()` API, instead of using the `.config("spark.sql.extensions", "")` api in which case `spark.sparkContext.getConf` won't be able to find `spark.sql.extensions` config. This is why I also had to try parsing a delta SQL command to see if the Delta SQL parser is available. We still want to do this right?

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 4 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:92]: We also have a case where a SparkSession is initialized with `.withExtensions()` API, instead of using the `.config("spark.sql.extensions", "")` api in which case `spark.sparkContext.getConf` won't be able to find `spark.sql.extensions` config. `DeltaSQLCommandTest` also uses this pattern. This is why I also had to try parsing a delta SQL command to see if the Delta SQL parser is available. We still want to do this right?

Comment (zsxwing) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:92]: Ah, I didn't know `withExtensions`. How about just checking  the `V2_SESSION_CATALOG_IMPLEMENTATION` config? I haven't seen any one who sets `V2_SESSION_CATALOG_IMPLEMENTATION` but forgets `spark.sql.extensions`.

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 2 above)
Comment (zsxwing) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:99]: We can change `def configureSparkSessionWithExtensionAndCatalog(originalException: Throwable)` to `def configureSparkSessionWithExtensionAndCatalog(originalException: Option[Throwable])` and call it.

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 4 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:92]: sounds good. Will just check for `V2_SESSION_CATALOG_IMPLEMENTATION` config.

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 9 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:569]: will do.

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 2 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:99]: Sounds good. Here's the implementation:
```scala
 def configureSparkSessionWithExtensionAndCatalog(
      originalException: Option[Throwable]): Throwable = {
    val catalogImplConfig = SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key
    new DeltaIllegalStateException(
      errorClass = "DELTA_CONFIGURE_SPARK_SESSION_WITH_EXTENSION_AND_CATALOG",
      messageParameters = Array(classOf[DeltaSparkSessionExtension].getName,
        catalogImplConfig, classOf[DeltaCatalog].getName),
      cause = originalException.getOrElse(null))
  }
```

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaErrorsSuite.scala (hunk 10 above)
Comment (ganeshchand) [reply core/src/test/scala/org/apache/spark/sql/delta/DeltaErrorsSuite.scala:50]: will revert back the changes.

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala [hunk 14]
```diff
@@ -1544,6 +1544,16 @@ trait DeltaErrorsBase
     )
   }
 
+  def configureSparkSessionWithExtensionAndCatalog(
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala:1547]: Could you remove the above `def configureSparkSessionWithExtensionAndCatalog(originalException: Throwable): Throwable` and fix existing usages? We don't need to keep two versions of such method.

File: core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala [hunk 15]
```diff
@@ -810,6 +810,15 @@ trait DeltaSQLConfBase {
           |""".stripMargin)
       .booleanConf
       .createWithDefault(false)
+
+  val DELTA_CHECK_REQUIRED_SPARK_CONF =
+    buildConf("checkRequiredSparkConf.enabled")
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala:815]: nit: `requiredSparkConfsCheck.enabled` to follow existing config key style such as `alterLocation.bypassSchemaCheck` and `retentionDurationCheck.enabled`.
`DELTA_CHECK_REQUIRED_SPARK_CONF` -> `DELTA_REQUIRED_SPARK_CONFS_CHECK`

File: core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala [hunk 16]
```diff
@@ -810,6 +810,15 @@ trait DeltaSQLConfBase {
...
+      .doc(
+      """Whether we check if SparkSession is initialized with spark.sql.extensions
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala:817]: `Whether to verify SparkSession is initialized with required configurations.` so that we don't need to change this if we add a new required configuration in future.

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaDataFrameHadoopOptionsSuite.scala [hunk 17]
```diff
@@ -29,6 +29,8 @@ class DeltaDataFrameHadoopOptionsSuite extends QueryTest with SQLTestUtils with
 
   protected override def sparkConf =
     super.sparkConf.set("spark.delta.logStore.fake.impl", classOf[LocalLogStore].getName)
+      // disable the spark conf check
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/sql/delta/DeltaDataFrameHadoopOptionsSuite.scala:32]: Instead of using this config, could you extend `DeltaSQLCommandTest` instead when it's working? Same for other tests. This config is added mostly for safety.

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala [hunk 18]
```diff
@@ -81,6 +82,20 @@ class DeltaLog private(
...
 
+  /**
+   * Verify the required Spark conf for delta
+   * Throw `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if
+   * `spark.sql.catalog.spark_catalog` config is missing. We do not check for
+   * `spark.sql.extensions` because DeltaSparkSessionExtension can alternatively
+   * be activated using the `.withExtension()` API. This check can be disabled
+   * by setting DELTA_CHECK_REQUIRED_SPARK_CONF to false.
+   */
+  if(spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_CHECK_REQUIRED_SPARK_CONF)) {
```
Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala:93]: could you add a new method for this code such as:
```
/**
...
*/
protected def checkRequiredConfigurations() {
 if(spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_CHECK_REQUIRED_SPARK_CONF)) {
    if (spark.sparkContext.conf.getOption(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key).isEmpty) {
      throw DeltaErrors.configureSparkSessionWithExtensionAndCatalog(None)
    }
  }
}

checkRequiredConfigurations()
```

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala (hunk 14 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala:1547]: Sure. I found two usages of this method. Will remove this method and refactor the existing function calls to use `configureSparkSessionWithExtensionAndCatalog(originalException: Option[Throwable]) instead.

Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala:1547]: Actually, `configureSparkSessionWithExtensionAndCatalog(originalException: Throwable)` throws `DeltaAnalysisException` and `configureSparkSessionWithExtensionAndCatalog(originalException: Option[Throwable])` throws `DeltaIllegalStateException`

Are we okay throwing `DeltaIllegalStateException` from `AnalysisHelper.isExtensionOrCatalogError()`?

Comment (zsxwing) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala:1547]: Let's use `DeltaAnalysisException` instead. Mostly of time when we create DeltaLog for the first time, it will be in the analysis phase.

Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala:1547]: I tried using `configureSparkSessionWithExtensionAndCatalog(originalException: Option[Throwable])` that throws `DeltaAnalysisException` and if the check if the spark config not found, it actually fails with `java.util.concurrent.ExecutionException` which is thrown by the `getDeltaLogFromCache()` called from the `DeltaLog` constructor. 

Because of this my unit test fails with:
```
org.scalatest.exceptions.TestFailedException: Expected exception org.apache.spark.sql.delta.DeltaAnalysisException to be thrown, but java.util.concurrent.ExecutionException was thrown.

Caused by: java.util.concurrent.ExecutionException: org.apache.spark.sql.delta.DeltaAnalysisException: This Delta operation requires the SparkSession to be configured with the
DeltaSparkSessionExtension and the DeltaCatalog. Please set the necessary
configurations when creating the SparkSession as shown below.

  SparkSession.builder()
    .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension")
    .config("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog")
    ...
    .getOrCreate()

```
Any idea how to make `intercept[DeltaAnalysisException] ` work in this case?

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 3 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala]: Apparently, `DeltaSQLCommandTest` sets the catalog config using `session.conf.set()` api as shown below:

```
 session.conf.set(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key, classOf[DeltaCatalog].getName)
```
if we use `spark.sparkContext.getConf()` api to check the config, then test classes that extend `DeltaSQLCommandTest` will fail

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaDataFrameHadoopOptionsSuite.scala (hunk 17 above)
Comment (ganeshchand) [reply core/src/test/scala/org/apache/spark/sql/delta/DeltaDataFrameHadoopOptionsSuite.scala]: I refactored this test class `DeltaDataFrameHadoopOptionsSuite.scala` to extend `DeltaSQLCommandTest` and all tests ran successfully. However, the question I have is that do we need all Delta test class to extend `DeltaSQLCommandTest` even if they actually don't need the SQL Parser and the catalog config (e.g. `DeltaLog`). When I refactored the test classes, my approach was - if the test class doesn't already extend `DeltaSQLCommandTest`, then set the boolean flag to skip the spark conf check, therefore not requiring the `DeltaSQLCommandTest` trait. Let me know if this makes sense.

File: core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala (hunk 18 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala]: done

File: core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala (hunk 15 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala]: done

File: core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala (hunk 16 above)
Comment (ganeshchand) [reply core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala]: done.

**Review by zsxwing — COMMENTED** [review]
Overall looks good. Could you also improve `DELTA_CONFIGURE_SPARK_SESSION_WITH_EXTENSION_AND_CATALOG` by providing the command to start spark-shell?

```
If you set the configurations via spark-shell/pyspark/spark-submit, you can add the following configurations to the command:

--conf spark.sql.extensions=<sparkSessionExtensionName> --conf <catalogKey>=<catalogClassName>
```

**Comment by zsxwing** [comment]
The changes look good to me. Could you resolve the conflicts?

**Comment by ganeshchand** [comment]
@zsxwing I have resolved the conflicts.

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaHistoryManagerSuite.scala [hunk 19]
```diff
@@ -47,6 +48,11 @@ trait DeltaTimeTravelTests extends QueryTest
     duration.toMillis
   }
 
+  protected override def sparkConf = {
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/sql/delta/DeltaHistoryManagerSuite.scala:51]: The change in this file is not needed since the test suite extends `DeltaSQLCommandTest`.

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaRestartSessionSuite.scala [hunk 20]
```diff
@@ -18,19 +18,24 @@ package org.apache.spark.sql.delta
...
 import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
 
 class DeltaRestartSessionSuite extends SparkFunSuite {
 
   test("restart Spark session should work") {
     withTempDir { dir =>
-      var spark = SparkSession.builder().master("local[2]").getOrCreate()
+      var spark = SparkSession.builder().master("local[2]")
+        .config(DeltaSQLConf.DELTA_REQUIRED_SPARK_CONFS_CHECK.key, "false")
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/sql/delta/DeltaRestartSessionSuite.scala:28]: Could you set the catalog rather than disabling the check?

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaRestartSessionSuite.scala [hunk 21]
```diff
@@ -18,19 +18,24 @@ package org.apache.spark.sql.delta
...
+        .getOrCreate()
       try {
         val path = dir.getCanonicalPath
         spark.range(10).write.format("delta").mode("overwrite").save(path)
         spark.read.format("delta").load(path).count()
 
         spark.stop()
-        spark = SparkSession.builder().master("local[2]").getOrCreate()
+        spark = SparkSession.builder().master("local[2]")
+          .config(DeltaSQLConf.DELTA_REQUIRED_SPARK_CONFS_CHECK.key, "false")
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/sql/delta/DeltaRestartSessionSuite.scala:37]: Could you set the catalog rather than disabling the check?

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaTimeTravelSuite.scala [hunk 22]
```diff
@@ -46,6 +48,11 @@ class DeltaTimeTravelSuite extends QueryTest
     duration.toMillis
   }
 
+  protected override def sparkConf = {
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/sql/delta/DeltaTimeTravelSuite.scala:51]: The change in this file is not needed since the test suite extends `DeltaSQLCommandTest`.

File: core/src/test/java/org/apache/spark/sql/delta/MergeIntoJavaSuite.java [hunk 23]
```diff
@@ -42,6 +42,8 @@ public class MergeIntoJavaSuite implements Serializable {
     public void setUp() {
         spark = new TestSparkSession();
         tempPath = Utils.createTempDir(System.getProperty("java.io.tmpdir"), "spark").toString();
+        // disable the spark conf check
+        spark.sqlContext().conf().setConfString("spark.databricks.delta.requiredSparkConfsCheck.enabled", "false");
```
Comment (zsxwing) [inline core/src/test/java/org/apache/spark/sql/delta/MergeIntoJavaSuite.java:46]: Could you set the catalog rather than disabling the check?

**Review by zsxwing — COMMENTED** [review]
Sorry for the delay. Could you add `with DeltaSQLCommandTest` to failed test suites rather than disabling the check?

File: core/src/test/java/org/apache/spark/sql/delta/MergeIntoJavaSuite.java (hunk 23 above)
Comment (ganeshchand) [reply core/src/test/java/org/apache/spark/sql/delta/MergeIntoJavaSuite.java]: I have reverted the changes. My local upstream/master was didn't have the latest version at the time I made this change. I have now reverted the changes.

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaHistoryManagerSuite.scala (hunk 19 above)
Comment (ganeshchand) [reply core/src/test/scala/org/apache/spark/sql/delta/DeltaHistoryManagerSuite.scala]: I have reverted the changes. My local upstream/master was didn't have the latest version at the time I made this change. I have now reverted the changes.

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaRestartSessionSuite.scala (hunk 20 above)
Comment (ganeshchand) [reply core/src/test/scala/org/apache/spark/sql/delta/DeltaRestartSessionSuite.scala]: done

File: core/src/test/scala/org/apache/spark/sql/delta/DeltaRestartSessionSuite.scala (hunk 21 above)
Comment (ganeshchand) [reply core/src/test/scala/org/apache/spark/sql/delta/DeltaRestartSessionSuite.scala]: done

**Review by zsxwing — APPROVED** [review]
LGTM!
